crates/sui-core/src/scoring_decision.rs

```diff
@@ -0,0 +1,225 @@
+        final_of_schedule: true,
+    };
+
+    update_low_scoring_authorities(&low_scoring, committee.clone(), reputation_scores);
+    assert_eq!(*low_scoring.load().get(&a4).unwrap(), 25_u64);
+    assert_eq!(low_scoring.load().len(), 1);
+
+    // a4 has score of 30 which is a bit lower, but not an outlier, so it should not be excluded
+    let mut scores = HashMap::new();
+    scores.insert(sec1.public().clone(), 45_u64);
+    scores.insert(sec2.public().clone(), 49_u64);
+    scores.insert(sec3.public().clone(), 55_u64);
+    scores.insert(sec4.public().clone(), 30_u64);
+    let reputation_scores = ReputationScores {
+        scores_per_authority: scores,
+        final_of_schedule: true,
+    };
+
+    update_low_scoring_authorities(&low_scoring, committee.clone(), reputation_scores);
+    assert_eq!(low_scoring.load().len(), 0);
+
+    // this set of scores has a high performing outlier, we don't exclude it
+    let mut scores = HashMap::new();
+    scores.insert(sec1.public().clone(), 45_u64);
+    scores.insert(sec2.public().clone(), 49_u64);
+    scores.insert(sec3.public().clone(), 55_u64);
+    scores.insert(sec4.public().clone(), 80_u64);
+    let reputation_scores = ReputationScores {
+        scores_per_authority: scores,
+        final_of_schedule: true,
+    };
+
+    update_low_scoring_authorities(&low_scoring, committee.clone(), reputation_scores);
+    assert_eq!(low_scoring.load().len(), 0);
+
+    // if more than the quorum is a low outlier, we don't exclude any authority
+    let mut scores = HashMap::new();
+    scores.insert(sec1.public().clone(), 45_u64);
+    scores.insert(sec2.public().clone(), 49_u64);
+    scores.insert(sec3.public().clone(), 16_u64);
+    scores.insert(sec4.public().clone(), 25_u64);
+    let reputation_scores = ReputationScores {
+        scores_per_authority: scores,
+        final_of_schedule: true,
+    };
+
+    update_low_scoring_authorities(&low_scoring, committee.clone(), reputation_scores);
+    assert_eq!(low_scoring.load().len(), 0);
+
+    // the computation can handle score values at any scale
+    let mut scores = HashMap::new();
+    scores.insert(sec1.public().clone(), 2300_u64);
+    scores.insert(sec2.public().clone(), 3000_u64);
+    scores.insert(sec3.public().clone(), 900_u64);
+    scores.insert(sec4.public().clone(), 1900_u64);
+    let reputation_scores = ReputationScores {
+        scores_per_authority: scores,
+        final_of_schedule: true,
+    };
+
+    update_low_scoring_authorities(&low_scoring, committee.clone(), reputation_scores);
+    assert_eq!(low_scoring.load().len(), 0);
+
+    // the computation can handle score values scaled up or down
+    // (note as we scale up sensitivity to outliers goes slightly down)
+    let mut scores = HashMap::new();
+    scores.insert(sec1.public().clone(), 2300_u64);
+    scores.insert(sec2.public().clone(), 3000_u64);
+    scores.insert(sec3.public().clone(), 210_u64);
+    scores.insert(sec4.public().clone(), 1900_u64);
```

Comment:
All the tests here are great! I still believe that we'll need to have a few more tests though with higher number of nodes (ex 40, 100) - I want to make sure that the parameters we have here do not overfit the case of for 4 nodes and can still work with the other cases. I understand the tuning the constant factors `K,C`  will eventually come from observation as well, but I would like to gain more confidence here.

Reply:
added 50 node cluster test